plugins/weather/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherReportWeek.java

```diff
@@ -23,44 +23,48 @@
 
 package de.unistuttgart.iaas.amyassist.amy.plugin.weather;
 
-import javax.xml.bind.annotation.XmlRootElement;
-import javax.xml.bind.annotation.XmlTransient;
-
-import com.github.dvdme.ForecastIOLib.FIODaily;
-
-import de.unistuttgart.iaas.amyassist.amy.utility.rest.Entity;
+import java.util.Arrays;
 
 /**
- * Serves as weather entity for current time
+ * Weather report for a week
  * 
- * @author Benno Krauß, Muhammed Kaya
+ * @author Benno Krauß, Muhammed Kaya, Tim Neumann
  */
-@XmlRootElement
-public class WeatherReportWeek extends Entity {
+public class WeatherReportWeek {
 
-	@XmlTransient
-	public final String preamble;
+	/** The reports for the days of the week. */
+	private final WeatherReportDay[] days;
+	/** The summary of the report. */
+	private final String summary;
 
-	public final WeatherReportDay[] days;
-	public final String summary;
+	/**
+	 * Creates a weather report for a week
+	 * 
+	 * @param pSummary
+	 *            Sets {@link #summary}
+	 * @param pDays
+	 *            Sets {@link #days}
+	 */
+	public WeatherReportWeek(String pSummary, WeatherReportDay[] pDays) {
+		this.summary = pSummary;
+		this.days = pDays;
+	}
 
 	/**
-	 * @param preamble
-	 *            Introduction text for speech
-	 * @param d
-	 *            FIODaily to get weather values of the week
+	 * Get's {@link #days days}
+	 * 
+	 * @return days
 	 */
-	public WeatherReportWeek(String preamble, FIODaily d) {
-		this.preamble = preamble;
-		this.days = new WeatherReportDay[d.days()];
-		for (int i = 0; i < d.days(); i++) {
-			this.days[i] = new WeatherReportDay(null, d.getDay(i));
-		}
-		this.summary = d.getSummary();
+	public WeatherReportDay[] getDays() {
+		return Arrays.copyOf(this.days, this.days.length);
```

Comment:
why copy the array here?

Reply:
So nobody can change it. The WeatherReportWeek should be immutable